add CONTRIBUTING.md (#4741)
official-stockfish/Stockfish: .github/CONTRIBUTING.md

Reviewer: I would suggest to not use any contractions, throughout the file. So `we're` to `we are` etc.

# Contributing to Stockfish

Welcome to the Stockfish project! We're excited that you're interested in contributing. This document outlines the guidelines and steps to follow when making contributions to Stockfish.

## Table of Contents

- [Getting Started](#getting-started)
  - [Setting up Stockfish](#setting-up-stockfish)
  - [Building Stockfish](#building-stockfish)
- [Making Contributions](#making-contributions)
  - [Reporting Issues](#reporting-issues)
  - [Submitting Pull Requests](#submitting-pull-requests)
- [Code Style](#code-style)
- [Community and Communication](#community-and-communication)
- [License](#license)

## Getting Started

### Setting up Stockfish

1. Fork this repository.
2. Clone your forked repository: `git clone https://github.com/your-username/Stockfish.git`
3. Add the Stockfish repository as a remote: `git remote add upstream https://github.com/official-stockfish/Stockfish.git`

### Building Stockfish

> In case you dont have a C++ Compiler installed you can follow the instructions from our wiki [linux](https://github.com/official-stockfish/Stockfish/wiki/Compiling-from-source#linux)/[windows](https://github.com/official-stockfish/Stockfish/wiki/Compiling-from-source#windows)/[macos](https://github.com/official-stockfish/Stockfish/wiki/Compiling-from-source#macos) to install one.

Reviewer 2: ```suggestion
In case you don't have a C++ compiler installed you can follow the instructions from our wiki [Linux](https://github.com/official-stockfish/Stockfish/wiki/Compiling-from-source#linux)/[Windows](https://github.com/official-stockfish/Stockfish/wiki/Compiling-from-source#windows)/[macOS](https://github.com/official-stockfish/Stockfish/wiki/Compiling-from-source#macos) to install one.
```

## Making Contributions

### Reporting Issues

If you find a bug, please open an issue on the [issue tracker](https://github.com/official-stockfish/Stockfish/issues). Be sure to include relevant information like your operating system, build environment, and a detailed description of the problem.

> _Please note that Stockfishs development isn't focused on adding new features. So any issues regarding missing features will potentially be closed without further discussion._

Reviewer 2: ```suggestion
_Please note that Stockfish's development isn't focused on adding new features. So any issue regarding missing features will potentially be closed without further discussion._
```

### Submitting Pull Requests

1. Make sure your forked repository is up to date with the official Stockfish repository: `git pull upstream master`

> 2. Create a new branch for your changes: `git checkout -b branch-name`

Reviewer 3: what about using the new syntax `git switch -c branch-name>` ?

Reviewer 4: I get this error:
```
    git switch -c my-branch  
    git: 'switch' is not a git command. See 'git --help'.
```

Reviewer 3: git 2.23.0 introduced in 2019 the switch command.

Author: changed to `switch`

3. Commit your changes and add a descriptive commit message.
   - If you are making a functional change, include the Bench in your commit message
     and submit it to fishtest. See [Creating my first test.](https://github.com/glinscott/fishtest/wiki/Creating-my-first-test#create-your-test)
   - Otherwise include "No functional change" in your commit message.
4. Push your branch to your forked repository: `git push --set-upstream origin branch-name`
5. Open a pull request against the `master` branch of the official Stockfish repository.  

>    This is easily done from your forked repository on GitHub. Github should prompt you to open a pull request when you push your branch to your forked repository.

Reviewer 2: ```suggestion
   This is easily done from your forked repository on GitHub. GitHub should prompt you to open a pull request when you push your branch to your forked repository.
```

_First time contributors should add their name to [AUTHORS](https://github.com/official-stockfish/Stockfish/blob/master/AUTHORS)._

> _Stockfish's development isn't focused on adding new features. So any pull requests introducing new features will potentially be closed without further discussion._

Reviewer 2: ```suggestion
_Stockfish's development isn't focused on adding new features. So any pull request introducing new features will potentially be closed without further discussion._
```

## Code Style

> We dont have a strict code style. But it's best to stick to the existing style of the file you are editing.

Reviewer 2: ```suggestion
We don't have a strict code style. But it's best to stick to the existing style of the file you are editing.
```
